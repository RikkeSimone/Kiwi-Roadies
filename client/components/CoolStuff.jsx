import React, { Component } from 'react'
import request from 'superagent'
import { connect } from 'react-redux'
import { addCampsites } from '../actions/campsites'
import { addHuts } from '../actions/huts'
// import { addTracks } from '../actions/tracks'
import proj4 from 'proj4'

const campURL = 'http://localhost:3000/api/v1/coolstuff/campsites'
const trackURL = 'http://localhost:3000/api/v1/coolstuff/tracks'
const hutURL = 'http://localhost:3000/api/v1/coolstuff/huts'

// EPSG 2193 NZGD2000 is the proj4string of the doc api data//
const tmerc = '+proj=tmerc +lat_0=0 +lon_0=173 +k=0.9996 +x_0=1600000 +y_0=10000000 +ellps=GRS80 +towgs84=0,0,0,0,0,0,0 +units=m +no_defs'
// input coords World Geodetic System 1984 (G1762) used in conversion, check this string below?
const wgs84 = '+proj=longlat +ellps=WGS84 +datum=WGS84 +no_defs'

class CoolStuff extends Component {
  state= {
    campsites: [],
    tracks: [],
    huts: []
  }

  componentDidMount () {
    request.get(campURL)
      .then(res => {
        const campsites = res.body
        // console.log('Long Lat:', proj4(tmerc, wgs84, [firstUTM.x, firstUTM.y]))
        campsites.forEach(campsite => {
          campsite.longLat = [campsite.x, campsite.y]
        })
        campsites.forEach(campsite => {
          campsite.longLat = proj4(tmerc, wgs84, campsite.longLat)
        })
        const action = addCampsites(campsites)
        this.props.dispatch(action)
      })
    request.get(hutURL)
      .then(res => {
        const huts = res.body
        // console.log('Long Lat:', proj4(tmerc, wgs84, [firstUTM.x, firstUTM.y]))
        huts.forEach(hut => {
          hut.longLat = [hut.x, hut.y]
        })
        huts.forEach(hut => {
          hut.longLat = proj4(tmerc, wgs84, hut.longLat)
        })
        console.log('CoolStuff Huts', huts)
        const action = addHuts(huts)
        this.props.dispatch(action)
      })
    request.get(hutURL)
      .then(res => {
        const huts = res.body
        // console.log('Long Lat:', proj4(tmerc, wgs84, [firstUTM.x, firstUTM.y]))
        huts.forEach(hut => {
          hut.longLat = [hut.x, hut.y]
        })
        huts.forEach(hut => {
          hut.longLat = proj4(tmerc, wgs84, hut.longLat)
        })
        console.log('CoolStuff Huts', huts)
        const action = addHuts(huts)
        this.props.dispatch(action)
      })
  }

  render () {
    return (
      <>
        <h2>Campsites</h2>
        <h2>{this.props.campsites[0] && this.props.campsites[0].name}</h2>
      </>
    )
  }
}

const mapStateToProps = state => {
  return {
    campsites: state.campsites,
    huts: state.huts
  }
}

export default connect(
  mapStateToProps
)(CoolStuff)
