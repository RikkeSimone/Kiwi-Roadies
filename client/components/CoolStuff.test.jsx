import React from 'react'
import { renderWithRedux } from '../testing/utils'
import '@testing-library/jest-dom'
import { screen } from '@testing-library/react'
import 'regenerator-runtime/runtime'

import CoolStuff from './CoolStuff'

jest.mock('./getCampsites', () => {
  return {
    getCampsites: function () {
      return Promise.resolve([
        { assetId: 100000811, name: 'LITTLE HELLFIRE CAMPSITE' },
        { assetId: 100030221, name: 'Lake Mackenzie Campsite' }
      ])
    }
  }
})

jest.mock('./getHuts', () => {
  return {
    getHuts: function () {
      return Promise.resolve([
        { assetId: 100000034, name: 'Bull Creek Hut' },
        { assetId: 100030976, name: 'Kellys Hunters Hut' }
      ])
    }
  }
})

jest.mock('./getTracks', () => {
  return {
    getTracks: function () {
      return Promise.resolve([
        { assetId: '1e72f18c-a04b-448b-95c5-2ce387948372', name: 'Te Iringa Track' },
        { assetId: 'b3e33c02-1732-4820-8715-4c7b9e337bd4', name: 'Arrowtown Chinese Settlement' }
      ])
    }
  }
})

test('CoolStuff has list of cool stuff to do', async () => {
  const { asFragment } = renderWithRedux(<CoolStuff/>)
  expect(asFragment()).toMatchSnapshot()
})

test('CoolStuff can take in an array from our Campsites API', async () => {
  renderWithRedux(<CoolStuff />)
  const post = await screen.findByText('LITTLE HELLFIRE CAMPSITE')
  expect(post).toBeInTheDocument()
})

test('CoolStuff can take in an array from our Huts API', async () => {
  renderWithRedux(<CoolStuff />)
  const post = await screen.findByText('Bull Creek Hut')
  expect(post).toBeInTheDocument()
})

test('CoolStuff can take in an array of items from our Tracks API', async () => {
  renderWithRedux(<CoolStuff />)
  const post = await screen.findByText('Te Iringa Track')
  expect(post).toBeInTheDocument()
})
