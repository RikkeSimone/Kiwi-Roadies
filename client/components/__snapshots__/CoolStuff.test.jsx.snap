// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoolStuff has list of cool stuff to do 1`] = `
<DocumentFragment>
  <h2>
    Campsites
  </h2>
  <h2 />
  <h2>
    Huts
  </h2>
  <h2 />
  <h2>
    Tracks
  </h2>
  <h2 />
</DocumentFragment>
`;
